app/assets/javascripts/app/discussions/discussion.js

```diff
@@ -29,7 +31,12 @@
       discussion.update();
     }
 
-    function toggleNewComment() {
+    function openEditForm(params) {
+      ctrl.comment = params.comment;
```

Comment:
You should be expecting the actual comment and not an object. It doesn't make much sense like this